Index: aws-lc/third_party/boringssl/CMakeLists.txt
===================================================================
--- aws-lc.orig/third_party/boringssl/CMakeLists.txt
+++ aws-lc/third_party/boringssl/CMakeLists.txt
@@ -19,6 +19,19 @@ enable_language(CXX)
 # run before all other targets.
 add_custom_target(global_target)
 
+if(UNIX AND NOT APPLE)
+    include(GNUInstallDirs)
+else()
+    set(CMAKE_INSTALL_LIBDIR "lib")
+    set(CMAKE_INSTALL_INCLUDEDIR "include")
+    set(CMAKE_INSTALL_BINDIR "bin")
+endif()
+
+install(DIRECTORY include/openssl
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        COMPONENT Development
+)
+
 if(ANDROID)
   # Android-NDK CMake files reconfigure the path and so Go and Perl won't be
   # found. However, ninja will still find them in $PATH if we just name them.
Index: aws-lc/third_party/boringssl/crypto/CMakeLists.txt
===================================================================
--- aws-lc.orig/third_party/boringssl/crypto/CMakeLists.txt
+++ aws-lc/third_party/boringssl/crypto/CMakeLists.txt
@@ -555,3 +555,10 @@ if(WIN32)
   target_link_libraries(crypto_test ws2_32)
 endif()
 add_dependencies(all_tests crypto_test)
+
+install(TARGETS crypto
+        EXPORT crypto-targets
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
+
Index: aws-lc/third_party/boringssl/ssl/CMakeLists.txt
===================================================================
--- aws-lc.orig/third_party/boringssl/ssl/CMakeLists.txt
+++ aws-lc/third_party/boringssl/ssl/CMakeLists.txt
@@ -62,3 +62,9 @@ if(WIN32)
   target_link_libraries(ssl_test ws2_32)
 endif()
 add_dependencies(all_tests ssl_test)
+
+install(TARGETS ssl
+        EXPORT ssl-targets
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
Index: aws-lc/third_party/boringssl/tool/CMakeLists.txt
===================================================================
--- aws-lc.orig/third_party/boringssl/tool/CMakeLists.txt
+++ aws-lc/third_party/boringssl/tool/CMakeLists.txt
@@ -37,3 +37,8 @@ else()
     target_link_libraries(bssl ssl crypto)
   endif()
 endif()
+
+install(TARGETS bssl
+        EXPORT bssl-targets
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
