Index: aws-lc/third_party/boringssl/CMakeLists.txt
===================================================================
--- aws-lc.orig/third_party/boringssl/CMakeLists.txt
+++ aws-lc/third_party/boringssl/CMakeLists.txt
@@ -80,6 +80,8 @@ if(NOT FIPS)
     # CMake automatically connects include_directories to the NASM
     # command-line, but not add_definitions.
     set(CMAKE_ASM_NASM_FLAGS "${CMAKE_ASM_NASM_FLAGS} -DBORINGSSL_DISPATCH_TEST")
+  elseif(CMAKE_BUILD_TYPE_LOWER MATCHES "rel")
+    add_definitions(-DBORINGSSL_RELEASE_BUILD)
   endif()
 endif()
 
Index: aws-lc/third_party/boringssl/crypto/cipher_extra/aead_test.cc
===================================================================
--- aws-lc.orig/third_party/boringssl/crypto/cipher_extra/aead_test.cc
+++ aws-lc/third_party/boringssl/crypto/cipher_extra/aead_test.cc
@@ -665,7 +665,7 @@ TEST_P(PerAEADTest, InvalidNonceLength)
   }
 }
 
-#if defined(SUPPORTS_ABI_TEST)
+#if defined(SUPPORTS_ABI_TEST) && !defined(BORINGSSL_RELEASE_BUILD)
 // CHECK_ABI can't pass enums, i.e. |evp_aead_seal| and |evp_aead_open|. Thus
 // these two wrappers.
 static int aead_ctx_init_for_seal(EVP_AEAD_CTX *ctx, const EVP_AEAD *aead,
Index: aws-lc/third_party/boringssl/crypto/test/abi_test.cc
===================================================================
--- aws-lc.orig/third_party/boringssl/crypto/test/abi_test.cc
+++ aws-lc/third_party/boringssl/crypto/test/abi_test.cc
@@ -208,7 +208,8 @@ template <typename... Args>
     WriteFile(stderr_handle, buf, strlen(buf), &unused, nullptr);
   }
 #else
-  write(STDERR_FILENO, buf, strlen(buf));
+  OPENSSL_UNUSED ssize_t unused_ret =
+          write(STDERR_FILENO, buf, strlen(buf));
 #endif
   abort();
 }
