Index: aws-lc/include/openssl/base.h
===================================================================
--- aws-lc.orig/include/openssl/base.h
+++ aws-lc/include/openssl/base.h
@@ -176,6 +176,8 @@ extern "C" {
 #define OPENSSL_VERSION_NUMBER 0x1010007f
 #define SSLEAY_VERSION_NUMBER OPENSSL_VERSION_NUMBER
 
+// BORINGSSL_API_VERSION is replaced with AWSLC_API_VERSION to avoid users interpreting AWSLC as BoringSSL.
+// Below are BorringSSL's comments on BORINGSSL_API_VERSION.
 // BORINGSSL_API_VERSION is a positive integer that increments as BoringSSL
 // changes over time. The value itself is not meaningful. It will be incremented
 // whenever is convenient to coordinate an API change with consumers. This will
@@ -184,7 +186,7 @@ extern "C" {
 // A consumer may use this symbol in the preprocessor to temporarily build
 // against multiple revisions of BoringSSL at the same time. It is not
 // recommended to do so for longer than is necessary.
-#define BORINGSSL_API_VERSION 10
+#define AWSLC_API_VERSION 10
 
 #if defined(BORINGSSL_SHARED_LIBRARY)
 
