diff --git a/third_party/fiat/p384_64.h b/third_party/fiat/p384_64.h
index 86dff393f..cd69adfdf 100644
--- a/third_party/fiat/p384_64.h
+++ b/third_party/fiat/p384_64.h
@@ -1728,6 +1728,7 @@ static void fiat_p384_add(uint64_t out1[6], const uint64_t arg1[6], const uint64
  * Output Bounds:
  *   out1: [[0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff]]
  */
+__attribute__((noinline))
 static void fiat_p384_sub(uint64_t out1[6], const uint64_t arg1[6], const uint64_t arg2[6]) {
   uint64_t x1;
   fiat_p384_uint1 x2;
