diff --git a/crypto/fipsmodule/evp/digestsign.c b/crypto/fipsmodule/evp/digestsign.c
index 6abd7722e..5354c79bf 100644
--- a/crypto/fipsmodule/evp/digestsign.c
+++ b/crypto/fipsmodule/evp/digestsign.c
@@ -129,6 +129,7 @@ int EVP_DigestVerifyInit(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
   return do_sigver_init(ctx, pctx, type, e, pkey, evp_verify);
 }
 
+__attribute__((noinline))
 int EVP_DigestSignUpdate(EVP_MD_CTX *ctx, const void *data, size_t len) {
   if (!uses_prehash(ctx, evp_sign)) {
     OPENSSL_PUT_ERROR(EVP, EVP_R_OPERATION_NOT_SUPPORTED_FOR_THIS_KEYTYPE);
