diff --git a/crypto/fipsmodule/aes/mode_wrappers.c b/crypto/fipsmodule/aes/mode_wrappers.c
index c4f5d49f3..aa770594d 100644
--- a/crypto/fipsmodule/aes/mode_wrappers.c
+++ b/crypto/fipsmodule/aes/mode_wrappers.c
@@ -68,6 +68,7 @@ static inline void aes_nohw_encrypt_wrapper(const uint8_t *in, uint8_t *out,
   aes_nohw_encrypt(in, out, key);
 }
 
+__attribute__((noinline))
 static inline void aes_hw_ctr32_encrypt_blocks_wrapper(const uint8_t *in,
                                                        uint8_t *out, size_t len,
                                                        const AES_KEY *key,
