diff --git a/crypto/fipsmodule/ec/ec.c b/crypto/fipsmodule/ec/ec.c
index 68ea8648d..591796288 100644
--- a/crypto/fipsmodule/ec/ec.c
+++ b/crypto/fipsmodule/ec/ec.c
@@ -1112,6 +1112,7 @@ int ec_point_mul_scalar(const EC_GROUP *group, EC_JACOBIAN *r,
   return 1;
 }
 
+__attribute__((noinline))
 int ec_point_mul_scalar_base(const EC_GROUP *group, EC_JACOBIAN *r,
                              const EC_SCALAR *scalar) {
   if (scalar == NULL) {
