diff --git a/tests/isc/Makefile.am b/tests/isc/Makefile.am
index 5cdd915..6ee1935 100644
--- a/tests/isc/Makefile.am
+++ b/tests/isc/Makefile.am
@@ -115,10 +115,12 @@ proxyheader_test_SOURCES =	\
 	proxyheader_test_data.h
 
 proxystream_test_CPPFLAGS =	\
-	$(AM_CPPFLAGS)
+	$(AM_CPPFLAGS)			\
+	$(OPENSSL_CFLAGS)
 
 proxystream_test_LDADD =	\
-	$(LDADD)
+	$(LDADD)				\
+	$(OPENSSL_LIBS)
 
 proxystream_test_SOURCES =	\
 	proxystream_test.c \
