diff --git a/src/test/ssl/t/002_scram.pl b/src/test/ssl/t/002_scram.pl
index dd93224124..44f570c8e2 100644
--- a/src/test/ssl/t/002_scram.pl
+++ b/src/test/ssl/t/002_scram.pl
@@ -155,14 +155,18 @@ $node->connect_ok(
 # Now test with a server certificate that uses the RSA-PSS algorithm.
 # This checks that the certificate can be loaded and that channel binding
 # works. (see bug #17760)
-if ($supports_rsapss_certs)
-{
-	switch_server_cert($node, certfile => 'server-rsapss');
-	$node->connect_ok(
-		"$common_connstr user=ssltestuser channel_binding=require",
-		"SCRAM with SSL and channel_binding=require, server certificate uses 'rsassaPss'",
-		log_like => [
-			qr/connection authenticated: identity="ssltestuser" method=scram-sha-256/
-		]);
-}
+#
+# AWS-LC does not support RSA-PSS certificates in libssl. If there is a relevant
+# feature request for this, cut an issue to our public repository.
+#
+# if ($supports_rsapss_certs)
+# {
+# 	switch_server_cert($node, certfile => 'server-rsapss');
+# 	$node->connect_ok(
+# 		"$common_connstr user=ssltestuser channel_binding=require",
+# 		"SCRAM with SSL and channel_binding=require, server certificate uses 'rsassaPss'",
+# 		log_like => [
+# 			qr/connection authenticated: identity="ssltestuser" method=scram-sha-256/
+# 		]);
+# }
 done_testing();
