diff --git a/testclients/mysql_client_test.cc b/testclients/mysql_client_test.cc
index 8bc55eda26..a37134221b 100644
--- a/testclients/mysql_client_test.cc
+++ b/testclients/mysql_client_test.cc
@@ -22324,6 +22324,9 @@ static void test_bug32915973() {
   mysql_stmt_close(stmt);
 }
 
+/* This test uses stateful session resumption, which is not supported in AWS-LC. */
+#if !defined (OPENSSL_IS_AWSLC)
+
 static void test_wl13075() {
   int rc;
   myheader("test_wl13075");
@@ -22456,6 +22459,7 @@ static void test_wl13075() {
     DIE_UNLESS(ret_ses_data == nullptr);
   }
 }
+#endif
 
 static void test_bug33535746() {
   DBUG_TRACE;
@@ -23770,7 +23774,9 @@ static struct my_tests_st my_tests[] = {
     {"test_bug32892045", test_bug32892045},
     {"test_bug33164347", test_bug33164347},
     {"test_bug32915973", test_bug32915973},
+#if !defined (OPENSSL_IS_AWSLC)
     {"test_wl13075", test_wl13075},
+#endif
     {"test_bug33535746", test_bug33535746},
     {"test_server_telemetry_traces", test_server_telemetry_traces},
     {"test_wl13128", test_wl13128},
