diff --git a/vio/viosslfactories.c b/vio/viosslfactories.c
index 530872c4..235bf76f 100644
--- a/vio/viosslfactories.c
+++ b/vio/viosslfactories.c
@@ -354,6 +354,11 @@ new_VioSSLFd(const char *key_file, const char *cert_file, const char *ca_file,
   }
 
   SSL_CTX_set_options(ssl_fd->ssl_context, ssl_ctx_options);
+  /* 
+    This reverts AWS-LC to the default OpenSSL auto-chaining behavior. Since 
+    OpenSSL enables auto-chaining by default, this is a no-op with OpenSSL.
+  */
+  SSL_CTX_clear_mode(ssl_fd->ssl_context, SSL_MODE_NO_AUTO_CHAIN);
 
   /*
     Set the ciphers that can be used
