diff --git a/Lib/test/libregrtest/worker.py b/Lib/test/libregrtest/worker.py
index 5d75bf7..bc1f643 100644
--- a/Lib/test/libregrtest/worker.py
+++ b/Lib/test/libregrtest/worker.py
@@ -1,7 +1,9 @@
+import signal
 import subprocess
 import sys
 import os
 from typing import Any, NoReturn
+from contextlib import contextmanager
 
 from test.support import os_helper, Py_DEBUG
 
@@ -18,6 +20,16 @@
     'test_ioctl',
 }
 
+@contextmanager
+def restore_sigint():
+    original_handler = signal.getsignal(signal.SIGINT)
+    signal.signal(signal.SIGINT, signal.default_int_handler)
+    try:
+        yield
+    except:
+        raise
+    finally:
+        signal.signal(signal.SIGINT, original_handler)
 
 def create_worker_process(runtests: WorkerRunTests, output_fd: int,
                           tmp_dir: StrPath | None = None) -> subprocess.Popen[str]:
@@ -128,4 +140,5 @@ def main() -> NoReturn:
 
 
 if __name__ == "__main__":
-    main()
+    with restore_sigint():
+        main()
